src/ListArray.java

```diff
@@ -0,0 +1,158 @@
+public class ListArray {
+
+    private int length;
```

Comment:
Be careful with naming - `length` means something very specific (think about `Array.length`).  There's a reason why Steve's demo code uses "size" here.